khelo/lib/ui/flow/tournament/add/add_tournament_view_model.dart

```diff
@@ -42,16 +42,36 @@ class AddTournamentViewNotifier extends StateNotifier<AddTournamentState> {
           currentUserId: userId,
         ));
 
+  TournamentModel? _editedTournament;
+
   void setUserId(String? userId) {
     state = state.copyWith(currentUserId: userId);
   }
 
+  void setData(TournamentModel? editedTournament) {
+    _editedTournament = editedTournament;
+    state.nameController.text = editedTournament?.name ?? '';
+    state = state.copyWith(
+      bannerImgUrl: editedTournament?.banner_img_url,
+      profileImgUrl: editedTournament?.profile_img_url,
+      startDate: editedTournament?.start_date ?? DateTime.now(),
+      endDate: editedTournament?.end_date ??
+          DateTime.now().add(const Duration(days: 1)),
+      selectedType: editedTournament?.type ?? TournamentType.knockOut,
+      teams: editedTournament?.teams ?? [],
+    );
+  }
+
   Future<void> onImageChange({
     required String imagePath,
     bool isBanner = false,
   }) async {
     try {
-      state = state.copyWith(imageUploading: true, actionError: null);
+      state = state.copyWith(
+        imageUploading: true,
+        actionError: null,
+        enableButton: _editedTournament != null,
```

Comment:
what if we remove title and then change image? isn't it will enable create button even if  tournament title is empty?